### PR TITLE
fix: cache the deployer to mitigate front-running on initialization

### LINKED ISSUE #145: Initialize() Owner is Frontrunnable
Currently, the owner of the manager and endpoint is set through whoever calls `initialize()`. Since the deployment of the contracts and call to initialize are separate, an attacker could frontrun this call to become the owner. 

By itself, we would probably just redeploy it after some sort of error occurs and we see that we're no longer the owner. However, I'm worried that there's a nice way to backdoor it (such as adding our own endpoint) then transfer ownership back to the user. 

There's a test demonstrating this under ``TestInitialize`` in ``tests/Upgrades.sol``. I think if we save the deployer then add a check in the call to initialize that the deployer is the caller then this should work fine.

### PR DESCRIPTION
**Pull request by RahulMaganti47**
Fixes #145.

### CODE REVIEW

File: evm/src/Manager.sol [hunk 1]
```diff
@@ -104,9 +107,15 @@ abstract contract Manager is
         mode = _mode;
         chainId = _chainId;
         evmChainId = block.chainid;
+        // save the deployer (check this on iniitialization)
+        deployer = _msgSender();
```
Comment (djb15) [inline evm/src/Manager.sol:111]: Any reason in particular why we're using the Context package here now but still msg.sender everywhere else?

**Review by djb15 — APPROVED** [review]

File: evm/src/Manager.sol (hunk 1 above)
Comment (RahulMaganti47) [reply evm/src/Manager.sol:111]: to cover the case in which the deployer is a contract to retrieve the address who initiated the deployment.

Comment (djb15) [reply evm/src/Manager.sol:111]: We don't override it anywhere though right so it's still `msg.sender`?

Comment (gator-boi) [reply evm/src/Manager.sol:111]: You're correct, I'll make the change.

**Review by djb15 — APPROVED** [review]

**Pull request merged**